### DISCUSSION TITLE
ckb 0.105.1 (e0dd357 2022-10-31)

### DISCUSSION THREAD

**Discussion by doitian** [nervosnetwork/ckb#3682] [Announcements]
This version fixed the conflicting options for light clien (#3665).

It also decreased the memory requirement to run the database migration (#3664).

### Compatability Table

| Network | Consensus Version        | Minimal CKB Version |
|---------|------------------------- | ------------------- |
| dev     | ckb2021 since epoch 0    |                     |
| testnet | ckb2021 since epoch 3113 | v0.101.0            |
| mainnet | ckb2021 since epoch 5414 | v0.103.0            |

See [how to connect to various chains](https://github.com/nervosnetwork/ckb/wiki/Chains) in Wiki.

### Downloads

| OS                | Arch  | Runtime Dependencies | Package                 | Sign                |
| ----------------- | ----- | -------------------- | ----------------------- | ------------------- |
| macOS             | x64   | macOS 10.15 or above | [zip][macos-zip]        | [PGP][macos-pgp]    |
| macOS  (Portable) | x64   | macOS 10.15 or above | [zip][macos-p-zip]      | [PGP][macos-p-pgp]  |
| Linux             | x64   | glibc, libstdc++     | [tarball][linux-zip]    | [PGP][linux-pgp]    |
| Linux  (Portable) | x64   | glibc, libstdc++     | [tarball][linux-p-zip]  | [PGP][linux-p-pgp]  |
| CentOS            | x64   | glibc, libstdc++     | [tarball][centos-zip]   | [PGP][centos-pgp]   |
| CentOS (Portable) | x64   | glibc, libstdc++     | [tarball][centos-p-zip] | [PGP][centos-p-pgp] |
| Windows           | x64   | VC++ Redistributable | [zip][windows-zip]      | [PGP][windows-pgp]  |
| Linux             | ARM64 | glibc, libstdc++     | [tarball][arm-zip]      | [PGP][arm-pgp]      |
| Docker            |       |                      | [dockerhub][]           |                     |

The `.asc` files are signatures. It is wise and more secure to check out for the files [integrity](https://github.com/nervosnetwork/ckb/blob/develop/docs/integrity-check.md).

Try the portable variants if the binaries complains about illegal instructions.

[macos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-apple-darwin.zip
[macos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-apple-darwin.zip.asc
[macos-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-apple-darwin-portable.zip
[macos-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-apple-darwin-portable.zip.asc
[linux-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-unknown-linux-gnu.tar.gz
[linux-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-unknown-linux-gnu.tar.gz.asc
[linux-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-unknown-linux-gnu-portable.tar.gz
[linux-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-unknown-linux-gnu-portable.tar.gz.asc
[centos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-unknown-centos-gnu.tar.gz
[centos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-unknown-centos-gnu.tar.gz.asc
[centos-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-unknown-centos-gnu-portable.tar.gz
[centos-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-unknown-centos-gnu-portable.tar.gz.asc
[windows-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-pc-windows-msvc.zip
[windows-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_x86_64-pc-windows-msvc.zip.asc
[arm-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_aarch64-unknown-linux-gnu.tar.gz
[arm-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.1/ckb_v0.105.1_aarch64-unknown-linux-gnu.tar.gz.asc
[dockerhub]: https://hub.docker.com/r/nervos/ckb/tags

<details><summary>Changes since v0.105.0</summary>

### Features

* #3672: Add rpc `estimate_cycles` (@zhangsoledad)
* #3650: Bump ckb-vm to v0.22.0 (@mohanson)
* #3664: Decrease max memory and increase speed, during chain root mmr migration (@yangby-cryptape)
* #3665: Remove `block_filter_enable` option (@quake)

    Two configuration options (`block_filter_enable` and `support_protocols`) are conflicting.

</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/nervosnetwork/ckb/releases/tag/v0.105.1'>ckb 0.105.1 (e0dd357 2022-10-31)</a>.</em>

**Comment by joshyates1980** [comment]
Do the developers of this release recommend migration or do the developers recommend to backup the old data, delete the data folder and synchronize again? I realize both will work, but I am only looking for what the developers suggest.

**Reply by doitian** [reply]
Backup and migrate. It is faster than re-sync on most machines.

**Comment by joshyates1980** [comment]
FYI to all. The migration will take a long time.

![image](https://user-images.githubusercontent.com/7255545/206262082-fa586f06-4f4a-42be-b4c1-8d2b078778db.png)

**Reply by joshyates1980** [reply]
Actually, the migration speeds up as it progresses; the (14h) was an estimate that has reduced dramatically an hour into the migration from the previous snapshot image: 

![image](https://user-images.githubusercontent.com/7255545/206273822-8f12dbd6-ec2b-4862-9f83-c3f4ea9ab57a.png)
